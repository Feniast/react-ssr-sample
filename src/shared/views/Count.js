import React from 'react';
import PropTypes from 'prop-types';
import Helmet from 'react-helmet';
import { connect } from 'react-redux';
import { countSelector, doubleCountSelector } from '../store/selectors/count';
import { increment, decrement, set } from '../store/actions/count';
import styles from '../styles/common.module.scss';

export class Count extends React.Component {
  static fetchData = (store) => {
    store.dispatch(set(~~(Math.random() * 100)));
  }

  static propTypes = {
    count: PropTypes.number.isRequired,
    doubleCount: PropTypes.number.isRequired,
    increment: PropTypes.func.isRequired,
    decrement: PropTypes.func.isRequired
  };

  render() {
    const { count, doubleCount, increment, decrement } = this.props;
    return (
      <div>
        <Helmet title="Count"></Helmet>
        <h1 className={styles.title}>Count</h1>
        <div>
          <p className="count">Count: {count}</p>
          <p>Double Count: {doubleCount}</p>
        </div>
        <button className="btn-plus" onClick={increment}>+</button>
        <button className="btn-minus" onClick={decrement}>-</button>
      </div>
    );
  }
}

export default connect(
  state => ({
    count: countSelector(state),
    doubleCount: doubleCountSelector(state)
  }),
  dispatch => ({
    increment: () => dispatch(increment()),
    decrement: () => dispatch(decrement())
  })
)(Count);
