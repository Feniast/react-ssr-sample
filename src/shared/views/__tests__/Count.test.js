import React from 'react';
import { shallow } from 'enzyme';
import { Count } from '../Count';

describe('About Views Test', () => {
  it('should render correctly', () => {
    const increment = jest.fn();
    const decrement = jest.fn();
    const count = 1;
    const component = shallow(
      <Count count={count} doubleCount={count * 2} increment={increment} decrement={decrement} />
    );

    expect(component.find('.count').text()).toEqual("Count: 1");

    component.find('.btn-plus').simulate('click');
    expect(increment).toHaveBeenCalled();

    component.find('.btn-minus').simulate('click');
    expect(decrement).toHaveBeenCalled();

    expect(component).toMatchSnapshot();
  });
});
