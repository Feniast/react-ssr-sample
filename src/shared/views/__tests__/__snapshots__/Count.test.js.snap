// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About Views Test should render correctly 1`] = `
<div>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
    title="Count"
  />
  <h1
    className="title"
  >
    Count
  </h1>
  <div>
    <p
      className="count"
    >
      Count: 
      1
    </p>
    <p>
      Double Count: 
      2
    </p>
  </div>
  <button
    className="btn-plus"
    onClick={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
  >
    +
  </button>
  <button
    className="btn-minus"
    onClick={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
  >
    -
  </button>
</div>
`;
